### PR TITLE
ProjectEuler | Update Problem statement of 002 to match projecteuler.net

### PR DESCRIPTION
**Pull request by sidtharthanan**
From projecteuler.net:
By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.
Correct: take all f(n) where f(n) <= 4,000,000 and f(n) is even
Incorrect: take all f(n) where n <= 4,000,000 and f(n) is even

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-2-even-fibonacci-numbers.english.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ forumTopicId: 301838
 <section id='description'>
 Each new term in the Fibonacci sequence is generated by adding the previous two terms. By starting with 1 and 2, the first 10 terms will be:
 <div style='text-align: center;'>1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...</div>
-By considering the terms in the Fibonacci sequence that do not exceed the <code>n</code>th term, find the sum of the even-valued terms.
+By considering the terms in the Fibonacci sequence that do not exceed number <code>n</code>, find the sum of the even-valued terms.
```
Comment (moT01) [inline curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-2-even-fibonacci-numbers.english.md]:
```suggestion
By considering the terms in the Fibonacci sequence whose values do not exceed <code>n</code>, find the sum of the even-valued terms.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
In the future, you should try and give your branch a better name. And it's a good idea to create an issue first before making a PR so it can be discussed.

You are correct, the way [projecteuler.net](https://projecteuler.net/problem=2) is different from our version.

**Comment by moT01** [comment]
Hey @RandellDawson, what do you think of changing this around? The forum guide would need to be updated, and older posts related to this challenge would no longer align. I'm fine with it since this is how it is on their site.

**Comment by RandellDawson** [comment]
@moT01 I had noticed the difference myself a long time ago.  I figured there was a reason they were different and never questioned it.  @scissorsneedfoodtoo What do you think?

**Comment by scissorsneedfoodtoo** [comment]
@RandellDawson, my understanding was that parts of the text was changed to accommodate for the different tests we have and the way the functions are written.

@moT01, thank you for making that suggestion. Normally I would be for sticking closer to the source material, but like you mentioned, we would have to make a lot of other changes. 

If we decide to change the text to "By considering the terms in the Fibonacci sequence whose values do not exceed <code>n</code>, find the sum of the even-valued terms," I believe we would also have to change the function and tests so the value of the `n`th term of the Fibonacci sequence is passed to `fiboEvenSum`. So our tests would call `fiboEvenSum(89)`, `fiboEvenSum(4181)`, `fiboEvenSum(46368)`, and so on.

While I think it's better we stick with the implementation we have, I could see updating the text if it's unclear. What do you both think?

**Comment by moT01** [comment]
I think the concept of the challenge is pretty much the same either way.

> If we decide to change the text, I believe we would also have to change the function and tests

Looks like the changes on this PR change the tests to account for this format.

> While I think it's better we stick with the implementation we have, I could see updating the text if it's unclear. What do you both think?

I don't think the text is unclear. The issue is simply that our version of the challenge is different from the original. Ours checks for < the `n`th term - and the original checks for < the `n`th value (of the term).

I really don't have an opinion on which way to go here. Either way works for me. The forum guide wouldn't be too tough to change. If there is no other reason to keep it like it is, I guess I would vote to change it to match the original.

**Comment by sidtharthanan** [comment]
> In the future, you should try and give your branch a better name. And it's a good idea to create an issue first before making a PR so it can be discussed.
> 
> You are correct, the way [projecteuler.net](https://projecteuler.net/problem=2) is different from our version.

I gone through the contribution guide and felt it is a small change that does not require any tests. so thought that an Issue is not required. But, in future I will remember to create one anyways.
I used Github UI to fork,commit and create a PR. So, sorry about the autogenerated branch name.
And I've accepted the changes suggested. Please review again and let me know if there is anything more.

**Comment by scissorsneedfoodtoo** [comment]
@sidtharthanan and @moT01, sorry about that! I didn't notice that the tests were already updated in my last review.

In that case, let's go with these changes since it brings the test and implementation closer to the original problem.

**Comment by sidtharthanan** [comment]
image

I don't understand this. Could you please explain if I still have to address any changes?

**Comment by scissorsneedfoodtoo** [comment]
@sidtharthanan, no need to worry. That just means that we need to officially approve your PR. It seems like you've made the changes suggested by @moT01, so I'll approve the PR now.

Once another reviewer confirms everything looks good we'll merge your changes.

**Review by scissorsneedfoodtoo — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM @sidtharthanan 🎉  Congratulations on your first contribution to the freeCodeCamp codebase 🎉 We look forward to future contributions. This will be merged shortly.

@scissorsneedfoodtoo, are we fine to merge this ahead of https://github.com/freeCodeCamp/freeCodeCamp/pull/38016? Looks like there will be some minor conflicts.

@RandellDawson, any chance you want to update [the guide](https://www.freecodecamp.org/forum/t/freecodecamp-challenge-guide-problem-2-even-fibonacci-numbers/301838) when this gets merged?
I made a quick solution if you want to use it at all...
```
function fiboEvenSum(n) {
  if (n <= 1) {
    return 0;
  } else {
    let nMinus2 = 1, nMinus1 = 2, value = 3, sum = 2;
    while(value < n) {  
      if(value % 2 === 0) {
        sum += value;
      }

      nMinus2 = nMinus1;
      nMinus1 = value;
      value = nMinus1 + nMinus2;
    }
    return sum;
  }
}
```

**Comment by RandellDawson** [comment]
>any chance you want to update the guide when this gets merged?

@moT01 I will take care of updating it once it is merged.

**Comment by moT01** [comment]
I think I'll just merge it, there's already conflicts on that other PR.

Thanks again @sidtharthanan

**Pull request merged**